DeepSpeech.ipynb

```diff
@@ -1462,9 +1431,24 @@
    "source": [
     "# Define CPU as device on which the muti-gpu testing is orchestrated\n",
     "with tf.device('/cpu:0'):\n",
-    "    # Test network\n",
-    "    test_decodings, test_labels, test_total_losses = decode_batch(data_sets.test)\n",
-    "    _, test_wer = calculate_and_print_wer_report(session, \"Test\", test_decodings, test_labels, test_total_losses)"
+    "    # Validation step\n",
+    "    originals = []\n",
+    "    results = []\n",
+    "    losses = []\n",
+    "    \n",
+    "    # Loop over the batches\n",
+    "    for batch in range(int(ceil(float(data_sets.test.total_batches) / dc))):\n",
+    "        result = session.run(get_report_params(data_sets.test), feed_dict)\n",
```

Comment:
Partially reverted this code to the former state and am now keeping all top paths together/alive throughout graph. They are going untouched through argument array nesting of ```session.run```. Also moved the current ```decoded[0]``` defaulting into a central location just before the report gets created. This should run-time wise be late enough for all sorts of cheap code changes.